lib/configurator.js

```diff
@@ -71,7 +79,14 @@ exports.configure = function configure (options) {
   if (!this.client.endpoint) {
     this.client.endpoint = 'https://' + this.client.host + this.client.pathname
   }
-
+  if (this.client.useServerSideCookie) {
+    if (
+      typeof options.cookieDomain === 'function' &&
+      this.client.useServerSideCookie
+    ) {
+      this.client.cookieDomain = this.client.cookieDomain(exports.DEFAULT_CONFIG.cookieDomain)
```

Comment:
@MikeMcElroy should we keep the original function instead of consuming it?

Reply:
Though if you wanted to, we could achieve both by actually converting passed strings into constant functions.